Source/WebCore/page/Navigation.cpp

```diff
@@ -211,51 +248,62 @@ Navigation::Result Navigation::navigate(const String& url, NavigateOptions&& opt
     }
 
     if (options.history == HistoryBehavior::Push && newURL.protocolIsJavaScript())
-        return createErrorResult(committed, finished, ExceptionCode::NotSupportedError, "A \"push\" navigation was explicitly requested, but only a \"replace\" navigation is possible when navigating to a javascript: URL."_s);
+        return createErrorResult(WTFMove(committed), WTFMove(finished), ExceptionCode::NotSupportedError, "A \"push\" navigation was explicitly requested, but only a \"replace\" navigation is possible when navigating to a javascript: URL."_s);
 
     if (options.history == HistoryBehavior::Push && currentURL.isAboutBlank())
-        return createErrorResult(committed, finished, ExceptionCode::NotSupportedError, "A \"push\" navigation was explicitly requested, but only a \"replace\" navigation is possible while on an about:blank document."_s);
+        return createErrorResult(WTFMove(committed), WTFMove(finished), ExceptionCode::NotSupportedError, "A \"push\" navigation was explicitly requested, but only a \"replace\" navigation is possible while on an about:blank document."_s);
 
     auto serializedState = serializeState(options.state);
     if (serializedState.hasException())
-        return createErrorResult(committed, finished, serializedState.releaseException());
+        return createErrorResult(WTFMove(committed), WTFMove(finished), serializedState.releaseException());
 
     if (!window()->protectedDocument()->isFullyActive())
-        return createErrorResult(committed, finished, ExceptionCode::InvalidStateError, "Invalid state"_s);
+        return createErrorResult(WTFMove(committed), WTFMove(finished), ExceptionCode::InvalidStateError, "Invalid state"_s);
 
     auto apiMethodTracker = maybeSetUpcomingNonTraversalTracker(WTFMove(committed), WTFMove(finished), WTFMove(options.info), serializedState.releaseReturnValue());
 
     // FIXME: This is not a proper Navigation API initiated traversal, just a simple load for now.
     frame()->loader().load(FrameLoadRequest(*frame(), newURL));
 
+    // If the load never made it to the navigate event.
     if (m_upcomingNonTraverseMethodTracker == apiMethodTracker) {
         // FIXME: Once the frameloader properly promotes the upcoming tracker with the navigate event `m_upcomingNonTraverseMethodTracker` should be unset or this will throw.
         m_upcomingNonTraverseMethodTracker = std::nullopt;
+        // Ref commitedPromise = apiMethodTracker.committedPromise.releaseNonNull();
+        // Ref finishedPromise = apiMethodTracker.finishedPromise.releaseNonNull();
+        // return createErrorResult(WTFMove(commitedPromise), WTFMove(finishedPromise), ExceptionCode::AbortError, "Navigation wasn't completed"_s);
```

Comment:
Is this safe to enable? If so I'd enable it, otherwise I'd remove it since the rule is to not include commented code.